shared/src/main/java/org/odk/collect/shared/injection/MultiClassProvider.kt

```diff
@@ -0,0 +1,25 @@
+package org.odk.collect.shared
```

Comment:
This is basically a 25 line mini DI framework. I think there are definitely better solutions than this but lets finish the discussion in #4940 before thinking about that.